### ISSUE TITLE
Add support for `ON JSON` index type in Search 

### ISSUE THREAD

**Issue by gkorland** [danitseitlin/redis-modules-sdk#93]

**Comment by MR4online** [comment]
Thanks @danitseitlin for your fast support!

I'm new with the combination of Redis JSON and Redis Search. (see my question over here: https://github.com/RedisJSON/RedisJSON/issues/381). I can not find "real" docs for this.
Aa far as I know: when creating an search index for Json data, we have to give it a name.
like:

FT.CREATE myIndexTest ON JSON PREFIX 1 test: SCHEMA $.name **AS name** TEXT $.cities AS cities TAG

Checking the code here https://github.com/danitseitlin/redis-modules-sdk/blob/3ffd5d263d5b9a2c01963c2bb57a50cd833fa86e/modules/redisearch.ts#L65
the name is immediately followed by the type.

So currently this will not work, because the name is missing:
`await client.create(index, "JSON", [{name: "$.name", type: "TEXT"}]);`
And this will not work, because it will set the "column name" as $.name AS name
`await client.create(index, "JSON", [{name: "$.name AS name", type: "TEXT"}]);`

To be clear: The Last statement will result the index definition to look like this:
FT.INFO i_2

 1) "index_name"
 2) "i_2"
 ...
 7) "fields"
 8) 1) 1) "$.name AS name"
       2) "type"
       3) "TEXT"
       4) "WEIGHT"
       5) "1"
...

And that is wrong.
I enabled debugging to understand whats been send via redis.sendCommand() or redis.send_command()

`  ioredis:redis write command[127.0.0.1:6380]: 0 -> FT.CREATE([ 'i_4',    'ON', 'JSON',   'SCHEMA', '$.name AS name',   'TEXT' ]) +6s`

But it should be:
`
  ioredis:redis write command[127.0.0.1:6380]: 0 -> FT.CREATE([ 'i_4',    'ON', 'JSON',   'SCHEMA', '$.name', 'AS', 'name',   'TEXT' ]) +6s
`

**in a nutshell**
For supporting RedisJson with RedisSearch (which would be awesome ;) ), we need an option to add a field alias name.
